velox/common/memory/tests/MappedMemoryTest.cpp

```diff
@@ -29,11 +31,19 @@ static constexpr uint64_t kMaxMappedMemory = 128UL * 1024 * 1024;
 static constexpr MachinePageCount kCapacity =
     (kMaxMappedMemory / MappedMemory::kPageSize);
 
-class MappedMemoryTest : public testing::Test {
+class MappedMemoryTest : public testing::TestWithParam<bool> {
  protected:
   void SetUp() override {
     auto tracker = MemoryUsageTracker::create(
         MemoryUsageConfigBuilder().maxTotalMemory(kMaxMappedMemory).build());
+    useMmap_ = GetParam();
+    if (useMmap_) {
+      MmapAllocatorOptions options = {kMaxMappedMemory};
+      mmapAllocator_ = std::make_unique<MmapAllocator>(options);
+      MappedMemory::setDefaultInstance(mmapAllocator_.get());
```

Comment:
A high-level comment. This change is quite large and complex. Before making this change, it would be nice to clean-up the existing code and remove MappedMemory::setDefaultInstance like we discussed.